### ISSUE TITLE
项目结构规划

### ISSUE THREAD

**Issue by wrp-wrp** [Wangyiyiwen/-#1]
<html>
<body>
<html><head></head><body><p>要把你刚才提到的几大模块——交互式 Chatbot、线下／线上换汇数据、汇率历史曲线展示、汇率预测模型——串联成一个完整的「最优换汇系统」，可以按下面思路来设计与实施。</p>
<hr>
<h2>一、总体架构</h2>
<pre><code>┌───────────────┐      ┌───────────────┐      ┌──────────────┐
│   用户界面/   │◀────▶│   Chatbot 层   │◀────▶│ 业务逻辑层    │
│   前端 App    │      │ (NLU/NLG+Dialog)│      │ (API 服务)   │
└───────────────┘      └───────────────┘      └─────┬────────┘
                                                      │
         ┌───────────────┐   ┌───────────────┐       │
         │ 汇率历史数据库 │   │ 线下渠道牌价  │◀──────┘
         │  （时序表）    │   │ （银行/机场等）│
         └───────────────┘   └───────────────┘
               ▲                       ▲
               │                       │
      ┌────────┴─────────┐     ┌───────┴────────┐
      │ 汇率可视化服务    │     │ 监控/抓取调度   │
      │ (曲线展示 + 报表) │     │ (Web Scraper)  │
      └──────────────────┘     └────────────────┘
               ▲
               │
        ┌──────┴───────┐
        │  汇率预测模型 │
        │  (LSTM/Prophet)│
        └───────────────┘
</code></pre>
<hr>
<h2>二、模块拆解与技术选型</h2>
<h3>1. Chatbot 层</h3>
<ul>
<li>
<p><strong>功能</strong>：理解用户意图（换汇目的地、金额、渠道偏好、时间要求等），并将请求转化为后端 API 调用。</p>
</li>
<li>
<p><strong>技术</strong>：</p>
<ul>
<li>
<p><strong>NLU/NLG</strong>：使用 Rasa / Microsoft Bot Framework / Dialogflow，做意图识别和槽位提取。</p>
</li>
<li>
<p><strong>对话管理</strong>：基于状态机或 Transformer+规则，维护用户上下文。</p>
</li>
</ul>
</li>
<li>
<p><strong>示例流程</strong>：</p>
<ol>
<li>
<p>用户：“我想在下周去新加坡，准备 2000 美金换成人民币，哪种方法最便宜？”</p>
</li>
<li>
<p>Chatbot 槽位填充：<code inline="">{ from: USD, to: CNY, amount:2000, date: next Monday, channels: all }</code></p>
</li>
<li>
<p>调用后端 <code inline="">/optimal-route</code> 接口拿结果，再用自然语言回复。</p>
</li>
</ol>
</li>
</ul>
<h3>2. 数据采集与存储</h3>
<ul>
<li>
<p><strong>线上标准汇率</strong>：</p>
<ul>
<li>
<p><strong>历史时序</strong>：Frankfurter / exchangerate.host 每日定时抓取，存到时序数据库（PostgreSQL / InfluxDB）。</p>
</li>
<li>
<p><strong>实时快照</strong>：每小时或更频繁刷新一行实时汇率。</p>
</li>
</ul>
</li>
<li>
<p><strong>线下渠道牌价</strong>：</p>
<ul>
<li>
<p><strong>Web Scraper</strong>：Python + BeautifulSoup/Requests，抓各大银行和机场官网的买入／卖出价。</p>
</li>
<li>
<p><strong>标准化存储</strong>：<code inline="">exchange_quotes(source, currency_pair, buy, sell, timestamp)</code>。</p>
</li>
</ul>
</li>
<li>
<p><strong>调度</strong>：Linux cron 或 APScheduler，失败发送报警。</p>
</li>
</ul>
<h3>3. 汇率可视化服务</h3>
<ul>
<li>
<p><strong>接口</strong>：<code inline="">GET /timeseries?base=CNY&amp;target=SGD&amp;start=…&amp;end=…</code> 返回 JSON。</p>
</li>
<li>
<p><strong>前端</strong>：React + Recharts/ECharts，或 Dash/Streamlit，绘制交互式时间序列图。</p>
</li>
<li>
<p><strong>后端</strong>：Flask/FastAPI，根据数据库查询并返回 DataFrame 序列。</p>
</li>
</ul>
<h3>4. 汇率预测模型</h3>
<ul>
<li>
<p><strong>数据准备</strong>：用历史时序构建训练集：<code inline="">features: window of past N 天汇率</code> → <code inline="">label: next-day汇率</code>。</p>
</li>
<li>
<p><strong>模型</strong>：</p>
<ul>
<li>
<p>简单有效：Facebook Prophet（季节 + 趋势）</p>
</li>
<li>
<p>深度学习：Bi-LSTM 或 Temporal Convolutional Network（TCN）</p>
</li>
</ul>
</li>
<li>
<p><strong>训练与部署</strong>：</p>
<ol>
<li>
<p>在 Python 环境（TensorFlow/PyTorch）训练，保存模型 <code inline="">model.pkl</code>。</p>
</li>
<li>
<p>部署到后端：API <code inline="">/forecast?base=CNY&amp;target=SGD&amp;period=30</code> 返回未来 30 天预测。</p>
</li>
<li>
<p>前端绘出预测曲线叠加在历史图上。</p>
</li>
</ol>
</li>
</ul>
<h3>5. 最优路径算法</h3>
<ul>
<li>
<p><strong>图构建</strong>：</p>
<ul>
<li>
<p><strong>节点</strong>：货币代码 × 渠道（可选）</p>
</li>
<li>
<p><strong>边</strong>：线上 “A→B” 权重＝−ln(汇率×(1−手续费))；线下 “A→B@渠道X” 权重＝−ln(buy_rate)</p>
</li>
</ul>
</li>
<li>
<p><strong>搜索</strong>：Dijkstra（无负环）或 Bellman–Ford（若含负环检测）。</p>
</li>
<li>
<p><strong>接口</strong>：<code inline="">GET /optimal-route?from=USD&amp;to=SGD&amp;amount=1000&amp;date=…&amp;channels=…</code></p>
</li>
<li>
<p><strong>输出</strong>：路径列表 + 总成本 + 每一步明细（哪个渠道、费率、手续费）。</p>
</li>
</ul>
<hr>
<h2>三、实施建议与分工</h2>

阶段 | 内容 | 建议负责人
-- | -- | --
需求细化 | 确定 Chatbot 要支持的意图和槽位 | 产品/你
Bot 开发 | 搭建 Rasa/Dialogflow 对话流程，测试用户交互 | 前端/AI
数据采集 | 编写并调度线上汇率和线下牌价抓取脚本，存入数据库 | 后端
Visualization | 后端 API + 前端图表展示（历史 & 预测） | 后端+前端
预测模型 | 数据准备、Prophet/LSTM 模型训练、模型评估与部署 | ML
最优算法 | 构建加权图、实现最短路径搜索、接口封装 | 算法
集成测试 | 端到端测试：从 Chatbot 问询到结果展示 | 全体
部署上线 | Docker/K8s 容器化，CI/CD，监控报警 | 运维

<hr>
<p>按照这个路线，你们就能把“人机交互”、“数据获取”、“可视化”、“预测”与“最优决策”五大部分串联起来，构建一个端到端的最优换汇系统。祝项目顺利！</p></body></html>
</body>
</html>

**Comment by wrp-wrp** [comment]
<img width="1342" height="1063" alt="Image" src="https://github.com/user-attachments/assets/772196e8-3043-42de-b1fc-2f0da472add6" />

**Comment by wrp-wrp** [comment]
https://app.diagrams.net/#G1jfK9dI2pNI3vXam5GNGM9TK_9DIihp7g#%7B%22pageId%22%3A%22cO8KAA9R90IackM79Wc7%22%7D
